climada/test/test_centr.py

```diff
@@ -83,6 +85,21 @@ def test_base_grid(self):
 
         self.assertEqual(count_sgi, 296)
 
+    def test_geodataframe(self):
+        """Test that constructing a valid Centroids instance from gdf works."""
+        gdf = gpd.GeoDataFrame(pd.read_excel(HAZ_TEMPLATE_XLS))
+        gdf.geometry = gpd.points_from_xy(
+                gdf['longitude'], gdf['latitude']
+        )
+        gdf['elevation'] = np.random.rand(gdf.geometry.size)
+
+        centroids = Centroids.from_geodataframe(gdf)
+        centroids.check()
+
+        self.assertEqual(centroids.geometry.size, 45)
+        self.assertEqual(centroids.elevation.size, 45)
+        self.assertIsInstance(centroids.geometry.total_bounds, np.ndarray)
+
```

Comment:
Maybe add a test for 
- when centrois.on_land.size == 0 (if this makes sense)
- Would it make sense to test the actual values in centroids.geometry and centroids.elevation instead of only there size?

Reply:
Since the test data does not set a region_id, this would have to take a synthetic region_id as input. I will write a test for the values in geometry, but not in elevation, since we're only copying a np.ndarray anyways; maybe checking the instance type would make more sense there.

Reply:
Added a few test cases; closing...